pkg/cmd/config/init/config_init.go

```diff
@@ -106,35 +138,38 @@ func NewConfigInitCmd(cfg *config.Config) *cobra.Command {
 	return newInitCmd
 }
 
-type p2pStageConfig struct {
-	DefaultEnvs []string `yaml:"default-envs"`
-}
-type initConfig struct {
-	Github struct {
-		Organization string `yaml:"organization"`
-	} `yaml:"github"`
-	Repositories struct {
-		Cplatform string `yaml:"cplatform"`
-		Templates string `yaml:"templates"`
-	} `yaml:"repositories"`
-	P2P struct {
-		FastFeedback p2pStageConfig `yaml:"fast-feedback"`
-		ExtendedTest p2pStageConfig `yaml:"extended-test"`
-		Prod         p2pStageConfig `yaml:"prod"`
-	} `yaml:"p2p"`
+func getConfigDirectory(cmd *cobra.Command, cfg *config.Config) string {
+	var directory string
+	initConfigDir, err := cmd.Flags().GetString("config-dir")
+	if err != nil {
+		logger.Panic().With(zap.Error(err)).Msg("could not get `--config-dir` flag")
+	}
+	// get environment variable for config dir
+	envConfigDir := os.Getenv("CORECTL_CONFIG_DIR")
+	if envConfigDir != "" && initConfigDir != "" {
+		logger.Panic().Msg("both --config-dir option and CORECTL_CONFIG_DIR environment variable are set, please use only one");
+	}
+	if initConfigDir != "" {
+		cfg.ConfigPaths.Directory.Value = initConfigDir
+		opt.ConfigDir = initConfigDir
+		fmt.Println("MARK: ConfigDir set to: ", initConfigDir)
+	}
+	return directory
 }
 
 func run(cmd *cobra.Command, opt *ConfigInitOpt, cfg *config.Config) error {
+	
+
 	// We don't allow the user to pass both the `--environments-repo` and the `--file` arguments
 	environmentsRepoFlagValue, err := cmd.Flags().GetString("environments-repo")
 	if err != nil {
 		logger.Panic().With(zap.Error(err)).Msg("could not get `--environments-repo` flag")
```

Comment:
this presents a problem, the logger will try to log to it's default file immediately, which hasn't been configured yet. Need to adjust so we know which directory we're logging to, and configure the logger first